src/utils/markdown_processor.py

```
@@ -0,0 +1,948 @@
+                # Current chunk is full, save it
+                chunks.append(
+                    {
+                        "content": "\n".join(current_chunk),
+                        "heading": current_heading,
+                        "size": current_size,
+                        "chunk_number": len(chunks),
+                    }
+                )
+
+                # Start new chunk with overlap
+                overlap_lines = current_chunk[-overlap // 50 :] if overlap > 0 else []
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)